### ISSUE TITLE
Add public method to create an instance of FetchedValue

### ISSUE THREAD

**Issue by swargolet** [graphql-java/graphql-java#3623]
v22.0 removed the public builder functions on `graphql.execution.FetchedValue` and with the constructor not being public either, there is no way to create an instance of FetchedValue.

We have our own ExecutionStrategy which overrides `unboxPossibleDataFetcherResult(...)`.  As that functions requires FetchedValue to be returned, we need to be able to make an instance of it for our use case where we have some logic to transform the fetched result based on various scenarios.

**Comment by bbakerman** [comment]
See https://github.com/graphql-java/graphql-java/pull/3624

**Comment by bbakerman** [comment]
Out of interest - what do you do in your custom unboxPossibleDataFetcherResult method?  What value processing do you do that needs change the value that was unboxed ?

**Comment by swargolet** [comment]
TY for making the change!  For now we are using reflection (I know, I know) to get an instance of FetchedValue.

As for our reasoning...
We have a custom annotation in which we can annotate any datafetcher with `@DefaultValue(...)` and provide a default value in the instance that the datafetcher returns null.  This is useful for us as we return either an empty optional or an empty list as opposed to null.  Only time null makes it back to `unboxPossibleDataFetcherResult` is if an error occurs.   This can be problematic for us for any non-nullable field.

Take for instance a schema like this:
```graphqls
type Query {
  movies: [Movie!]!
}

type Movie {
  title: String!
  genre: Genre!
  actors: [Actor!]!
}

type Genre {
...
}

type Actor {
  ...
}
```

If an error occurs while fetching genre and/or actors and that error bubbles up, it will be caught internally by the library and throw an error `cannot return null for non-nullable field`, this will cause the query itself then to error out as Movies is non-nullable and in the end even if title or some other field was fetched properly, the whole data response becomes null.   While it is easy enough to handle the error more gracefully in the datafetcher and return a DatafetcherResult, we wanted a more convenient way to do this across our codebase, so we made this custom annotation which we then can make use if it within the `unboxPossibleDataFetcherResult` function.  

I hope I explained it well.
